### PR TITLE
Refactor Code ✔✔

### PR DESCRIPTION
**Pull request by Talha-Hanif5666**
**In this PR we have done refactoring our code:**

- [ ] Refactor main.rb so that it look clean. ✔✔

- [ ] Make a new file, interface.rb for showing interface of app. ✔✔

### CODE REVIEW

File: rb-files/interface.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  puts '7 - Exit'
+end
+
+def handle_user_choice(choice, app)
+  menu_options = {
+    1 => :list_books,
+    2 => :list_people,
+    3 => :create_person,
+    4 => :create_book,
+    5 => :create_rental,
```
Comment (tugsem) [inline rb-files/interface.rb]: - [ ] It is a bad example of a single-responsibility rule. All requests for input and input validation in your methods should've been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input. Kindly remove them from your functions in the app.rb and move them to this file. You can check this lesson to remember them: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid
- 
image
image
image

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello  @Talha-Hanif5666  and @RaphDasilva 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️linters are passing
✔️correct use of gitflow
✔️professional documentation
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rb-files/interface.rb (hunk 1 above)
Comment (Talha-Hanif5666) [reply rb-files/interface.rb]: @tugsem I don't understand your comment. Please elaborate it..

File: rb-files/app.rb [hunk 2]
```diff
@@ -97,17 +84,19 @@ def create_rental
   def list_rentals_by_person_id
     print 'ID of person: '
     id = gets.chomp.to_i
```
Comment (mikethreels) [inline rb-files/app.rb]: - [ ] The previous collector refers to this part of the code, All requests for input and input validation in your methods should've been placed in the main.rb file and/or classes/methods that are responsible for requesting or validating input. 
which means that this part of the code should be moved to a class or method which specifically handles user input.

File: rb-files/app.rb [hunk 3]
```diff
@@ -23,72 +19,63 @@ def list_books
...
   end
 
   def create_person
-    print 'Do you want to Create a students (1) or a teacher (2)? [Input the number]: '
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
     select_person = gets.chomp.to_i
     print 'Name: '
     name = gets.chomp
     print 'Age: '
     age = gets.chomp.to_i
```
Comment (mikethreels) [inline rb-files/app.rb]: - [ ] The previous collector refers to this part of the code, All requests for input and input validation in your methods should've been placed in the main.rb file and/or classes/methods that are responsible for requesting or validating input. 
which means that this part of the code should be moved to a class or method which specifically handles user input.

File: rb-files/app.rb [hunk 4]
```diff
@@ -23,72 +19,63 @@ def list_books
...
+    puts 'Select a person from the following list by number (not ID)'
     @people.each_with_index do |person, index|
       puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
     end
-
     person_index = gets.chomp.to_i
-    puts
-
     print 'Date: '
     date = gets.chomp
```
Comment (mikethreels) [inline rb-files/app.rb]: - [ ] The previous collector refers to this part of the code, All requests for input and input validation in your methods should've been placed in the main.rb file and/or classes/methods that are responsible for requesting or validating input. 
which means that this part of the code should be moved to a class or method which specifically handles user input.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

### In case you need additional information please send a notification on slack as well, this might help speed up the response.

 ## Highlights
- Good job, reaching out to the CR for more information.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by BregornOriginal — APPROVED** [review]
### Hi @Talha-Hanif5666 

### Your project is complete!  There is nothing else to say other than... it's time to merge it 

### Congratulations! 



## Highlights

✅ Linters are passing
✅ Good commit history
✅ Good work refactoring your code 🏁 

[OPTIONAL] - 🥇  As you are writing a new application, it would be great to be more cleanest and readable as possible. Remember to remove commented code in your applications when you make a new Pull Request or ask for a review 🥇 

## Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.
Cheers and Happy coding!:aplauso::aplauso::aplauso:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

Please, remember to tag me in your question so I can receive the notification.
------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)[.](https://airtable.com/shrQAqnBwek5a0O0s)._)

**Pull request merged**